### ISSUE TITLE
mAP on evaluation server is all 0

### ISSUE THREAD

**Issue by SM1991CODES** [waymo-research/waymo-open-dataset#238]
Hi,

I generated predictions on a subset of the validation set and uploaded the file (after doing create_submission) to the eval server.
I get all 0s.
What could be wrong?

Does the eval server automatically take the corresponding subset of gt frames?

I tried offline eval using the downloaded gt from google storage, it also gives all 0s.

Please help me out.

Best Regards
Sambit

**Comment by peisun1115** [comment]
It is 0 usually because [these fields](https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/protos/metrics.proto#L39-L50) are not set properly. We require them to be matched exactly.

**Comment by SM1991CODES** [comment]
Thank you for the quick answer. But I have set these fields.
Here is what I do:
for every tfrecord, take every 10th frame (1s apart). Save the context and timestamps, predict.
Then I create a single write .bin object for all detected objects across all frames.

But your other comment about gt.bin means that since I am predicting only on about 500 frames, all others are FN and hence the score is nearly 0. Could that be the case?

Please confirm.

Best Regards
Sambit

**Comment by peisun1115** [comment]
i think it shouldn't be exactly 0 if you believe you are predicting anything meaningful for those 1/10 frames. Can you try the following:

1. Find a frame (identified by context_name, timestamp). Read its ground truth directly from python.
2. Open the bin file you produced. Print the predictions. Make sure "context_name, timestamp" match. And make sure there is any predicted box with IoU > 0.7.

If you still see AP = 0 after these 2 steps are validated. Then can you send me the "context_name, timestamp" you have checked, and the ground truth boxes, your predictions for that frame.

**Comment by SM1991CODES** [comment]
Hi,

Thank you for that advice.
Just one important point - I am predicting 3D objects only using the top
lidar.
So I am also writing labels for (x, y, z, l, w, h, heading) for that.
Hope that is not a problem, is it?
As in, do I need to convert them to camera coordinates?

I just read something about gt in camera frame and wanted to make sure.

Best Regards
Sambit

**Comment by peisun1115** [comment]
no. that shouldn't be a problem. You don't need to convert to camera frame. All predictions should be in vehicle frame which is what we use for ground truths.

**Comment by SM1991CODES** [comment]
Okay, thank you.
I am just struggling a lot with several topics.
I am trying to detect objects using range images.

Some of the topics I am struggling with are:
1) 8GB GPU memory on my workstation laptop is not enough to process even
(2, 1, 64, 2650) range images produced by waymo code.
2) Could not yet figure out how to download using gsutil - GCP compute
engine gives authentication error, no access for bucket.objects.list, I
tried several times and referred several posts but no result.
3) I tried splitting up each RI into 10 sections of 64, 256, thereby
ignoring the last 90 columns. This is the one which I was able to train and
predict on the validation set.
The visual results served as a proof of concept but then eval problem.
4) Now I think splitting up range images into several sections is not
really a good idea for the network. Also, I plan to use the intensity
channel.

Summarily, I just wanted to inform you of all the problems I face right
now. I totally understand you cannot help me with everything.

Best Regards
Sambit
